cmd/soroban-cli/src/commands/contract/extend.rs

```diff
@@ -222,9 +222,12 @@ impl NetworkRunnable for Cmd {
             let entry = client.get_full_ledger_entries(&keys).await?;
             let extension = entry.entries[0].live_until_ledger_seq;
 
-            if entry.latest_ledger + i64::from(extend_to) < i64::from(extension) {
+            let new_ext = entry.latest_ledger + i64::from(extend_to);
```

Comment:
I don't understand what this block tries to do. If changes are empty, then extension didn't happen. If extension didn't happen, then chances are we shouldn't have sent this transaction at all. There is only an edge case where someone extends the entry just before you, so that the transaction becomes no-op only at apply time. Even then, I think it's better to just say that transaction was a no-op and entry had sufficient TTL. But I think we shouldn't send the transaction when it's going to be no-op, similarly to what we need to do for the restore command.

Reply:
Yeah, this makes much more sense to me. As I've mentioned in the comment, we still need to look at the simulation results to avoid sending no-ops in the first place, but this block still makes sense for the edge cases. We can update the simulation parsing logic as a followup.

Reply:
> we still need to look at the simulation results to avoid sending no-ops in the first place

Makes sense 👍